modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -462,13 +463,43 @@ def merge(self, right, **kwargs):
             def map_func(left, right=right, kwargs=kwargs):
                 return pandas.merge(left, right, **kwargs)
 
+            new_columns = None
+            new_dtypes = None
+            if self._modin_frame._columns_cache is not None:
+                if left_on is None and right_on is None:
```

Comment:
```suggestion
                if left_index is False and right_index is False:
                    if left_on is None and right_on is None:
```

Reply:
we're already in this branch
https://github.com/modin-project/modin/blob/0df5fbf7a66663110878633467e4488ad37b0337/modin/core/storage_formats/pandas/query_compiler.py#L457